experimental/compactdecimal/src/compactdecimal.rs

```diff
@@ -147,44 +181,139 @@ impl CompactDecimalFormatter {
         })
     }
 
-    /// TODO(egg): Explain that this is an advanced API which can fail (and you
-    /// should use the API that does not exist instead).
+    icu_provider::gen_any_buffer_constructors!(
+        locale: include,
+        options: skip,
+        error: CompactDecimalError,
+        functions: [
+            Self::try_new_long_unstable,
+            try_new_long_with_any_provider,
+            try_new_long_with_buffer_provider
+        ]
+    );
+
+    /// Formats an integer in compact decimal notation using the default
+    /// precision settings.
     ///
-    /// # Examples
+    /// The result may have a fractional digit only if it is compact and its
+    /// significand is less than 10. Trailing fractional 0s are omitted.
     /// ```
-    /// use fixed_decimal::CompactDecimal;
-    /// use icu_compactdecimal::CompactDecimalFormatter;
-    /// use icu_decimal::options::FixedDecimalFormatterOptions;
-    /// use icu_locid::locale;
-    /// use writeable::assert_writeable_eq;
-    /// use std::str::FromStr;
+    /// # use icu_compactdecimal::CompactDecimalFormatter;
+    /// # use icu_locid::locale;
+    /// # use writeable::assert_writeable_eq;
+    /// #
+    /// # let short_english = CompactDecimalFormatter::try_new_short_unstable(
+    /// #    &icu_testdata::unstable(),
+    /// #    &locale!("en").into()
+    /// # ).unwrap();
+    /// assert_writeable_eq!(short_english.format(2), "2");
+    /// assert_writeable_eq!(short_english.format(843), "843");
+    /// assert_writeable_eq!(short_english.format(2207), "2.2K");
+    /// assert_writeable_eq!(short_english.format(15_127), "15K");
+    /// assert_writeable_eq!(short_english.format(3_010_349), "3M");
+    /// ```
+    /// The result is the nearest such compact number, with halfway cases-
+    /// rounded towards the number with an even least significant digit.
+    /// ```
+    /// # use icu_compactdecimal::CompactDecimalFormatter;
+    /// # use icu_locid::locale;
+    /// # use writeable::assert_writeable_eq;
+    /// #
+    /// # let short_english = CompactDecimalFormatter::try_new_short_unstable(
+    /// #    &icu_testdata::unstable(),
+    /// #    &locale!("en").into(),
+    /// # ).unwrap();
+    /// assert_writeable_eq!(short_english.format(999_499), "999K");
+    /// assert_writeable_eq!(short_english.format(999_500), "1M");
+    /// assert_writeable_eq!(short_english.format(1650), "1.6K");
+    /// assert_writeable_eq!(short_english.format(1750), "1.8K");
+    /// assert_writeable_eq!(short_english.format(1950), "2K");
+    /// ```
+    pub fn format(&self, value: i64) -> FormattedCompactDecimal<'_> {
+        let unrounded = FixedDecimal::from(value);
+        let log10_type = unrounded.nonzero_magnitude_start();
+        let (mut plural_map, mut exponent) = self.plural_map_and_exponent_for_magnitude(log10_type);
+        let mut significand = unrounded.multiplied_pow10(-i16::from(exponent));
+        if significand.nonzero_magnitude_start() == 0 {
```

Comment:
Issue: This behavior is slightly different than ICU and ECMA-402. In those implementations, we retain 2 significant digits all the way down. We should try to align.

```javascript
new Intl.NumberFormat("en", { notation: "compact" }).format(1.2e-5)
// '0.000012'
```

Reply:
1.2e-5 is not a valid input here, since we take `i64`s.
Note that this is looking at the magnitude of the significand on its own, not the actual magnitude of the number being formatted, so this condition is « if we show one digit before the decimal point » (and then that branch does « show one more after the decimal point »).

I will add some comments that gloss the logic like that so it is clearer.

Reply:
Yep, this isn't a problem since we only have integers. We shouldn't forget about it when supporting non-integers.